src/DL/TransactionHistoryHandler.java

```diff
@@ -0,0 +1,77 @@
+package DL;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+
+public class TransactionHistoryHandler {
+
+    private final String path;
+
+
+    public TransactionHistoryHandler(String path) {
+        create(path);
+        this.path = path;
+    }
+
+
+    private boolean create(String path) {
+        File file = new File(path);
+        try {
+            return file.createNewFile();
+        } catch (IOException e) {
+            System.out.println(e.getMessage());
+            return false;
+        }
+    }
+
+    public ArrayList<Transaction> read(){
+        ArrayList<Transaction> result = new ArrayList<>();
+        try(Scanner scanner = new Scanner(new File(this.path))){
+            while (scanner.hasNextLine()) {
+                String newTask = scanner.nextLine();
+                String[] propertiesOfTransaction = newTask.split(";");
+                Transaction transaction = new Transaction(
+                        convertingStringToDateTime(propertiesOfTransaction[0]),
+                        Double.parseDouble(propertiesOfTransaction[1].replace(",",".")),
+                        CurrencyTitle.valueOf(propertiesOfTransaction[2]),
+                        CurrencyTitle.valueOf(propertiesOfTransaction[3]),
+                        Double.parseDouble(propertiesOfTransaction[4].replace(",",".")));
+                result.add(transaction);
+            }
+            return result;
+        }catch (FileNotFoundException e){
+            System.out.println(e.getMessage());
+        }
+        return result;
+    }
+
+
+    public void append(Transaction transaction){
+        try (FileWriter fileWriter = new FileWriter(this.path,true)) {
+            fileWriter.append(convertedTransaction(transaction)+"\n");
```

Comment:
if you add "\n" for each line you will always have an empty line at the end of the file. and this will break your read method on the last loop iteration. 
there may be different approaches, for example, we can add a new line symbol at the beginning of the line if the file is not empty. for this we should have some private method isEmpty() which helps us to check file each time

Reply:
Done